flink-table/flink-sql-jdbc-driver/src/main/java/org/apache/flink/table/jdbc/FlinkResultSet.java

```diff
@@ -184,7 +185,7 @@ public int getInt(int columnIndex) throws SQLException {
         checkValidRow();
         checkValidColumn(columnIndex);
         try {
-            return currentRow.getInt(columnIndex - 1);
```

Comment:
According to the comment of `java.sql.ResultSet#getInt`, we need to adjust `null` to the corresponding value. This is the same for other primitive types.
```java
     * @return the column value; if the value is SQL <code>NULL</code>, the
     * value returned is <code>0</code>
```